File: api/DTOs/Book/BookDto.cs
namespace api.DTOs.Book
    {
    public class BookDto
        {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public string Description { get; set; }
        public int GenreId { get; set; }
        public string GenreName { get; set; }
        public string CoverImageUrl { get; set; }
        public DateTime PublishedDate { get; set; }
        public bool IsAvailable { get; set; }
        }
    }

File: client/src/app/shared/dtos/book.dto.ts
export interface BookDto {
  id: number;
  title: string;
  author: string;
  description: string;
  genreId: number;
  genreName: string;
  coverImageUrl: string;
  publishedDate: string;
  isAvailable: boolean;
}

export interface CreateBookDto {
  title: string;
  author: string;
  description?: string;
  genreId: number;
  coverImageUrl?: string;
  publishedDate: string;
}

export interface UpdateBookDto {
  title: string;
  author: string;
  description?: string;
  genreId: number;
  coverImageUrl?: string;
  publishedDate: string;
}

export interface BookFilterDto {
  title?: string;
  author?: string;
  genreId?: number;
  pageNumber: number;
  pageSize: number;
}
